pkg/expressions/partials.go

```diff
@@ -369,3 +378,60 @@ func CompareDestinationPortSet(set *nftables.Set) ([]expr.Any, error) {
 func CompareDestinationPortSetWithRegister(set *nftables.Set, reg uint32) ([]expr.Any, error) {
 	return []expr.Any{DestinationPort(reg), PortSetLookUp(set, reg)}, nil
 }
+
+func BitwiseWithRegisters(sourceRegister uint32, destRegister uint32, length uint32, mask []byte, xor []byte) *expr.Bitwise {
+	return &expr.Bitwise{
+		SourceRegister: sourceRegister,
+		DestRegister:   destRegister,
+		Len:            length,
+		Mask:           mask,
+		Xor:            xor,
+	}
+}
+
+func Bitwise(length uint32, mask []byte, xor []byte) *expr.Bitwise {
+	return BitwiseWithRegisters(defaultRegister, defaultRegister, length, mask, xor)
+}
+
+func LoadCtByKeyWithRegister(ctKey expr.CtKey, reg uint32) (expr.Ct, error) {
```

Comment:
Please add a validation check for `ctKey`. It seems like a range check might suffice, `unix.NFT_CT_STATE` to `unix.NFT_CT_EVENTMASK`.

https://github.com/google/nftables/blob/main/expr/ct.go#L31

Reply:
https://github.com/ngrok/firewall_toolkit/pull/16#discussion_r1157649642